tools/testsys/Test.toml.example

```diff
@@ -0,0 +1,99 @@
+# is used to validate Bottlerocket builds.
+
+# This section contains configuration details for all testing
+[test]
+# The repo from `Infra.toml` that should be used for testing. It may be
+# useful to create a repo in `Infra.toml` that contains the infrastructure
+# needed for testing
+repo = "default"
+# The registry containing alternate TestSys agent images
+testsys-images-registry = "public.ecr.aws/bottlerocket-test-system"
+
+# The URI for the EKS resource agent that should be used.
+# An individual agent's provided URI will be used even if `testsys-images-registry` is present.
+eks-resource-agent-image = "public.ecr.aws/bottlerocket-test-system/eks_resource_agent:v0.0.2"
+
+# The test configuration is broken up into different levels depending on the variant.
+# Configurations will have the following precedence:
+# Arguments from the command line
+# Environment variables from the command line
+# '<VARIANT>' configuration in `Test.toml`
+# '<FAMILY>-<FLAVOR>' configuration in `Test.toml`
+# '<FAMILY>' configuration in `Test.toml`
+# '<PLATFORM>-<FLAVOR>' configuration in `Test.toml`
+# '<PLATFORM>' configuration in `Test.toml`
+# TestSys default values
+
+# For the `aws-k8s-1.23-nvidia` variant on `x86_64`, configurations will have the following precedence:
+# Arguments from the command line
+# ['aws-k8s-1.23-nvidia'.x86_64] - [<VARIANT>.<ARCH>]
+# ['aws-k8s-1.23-nvidia'] - [<VARIANT>]
+# ['aws-k8s-nvidia'.x86_64] - [<FAMILY>-<FLAVOR>.<ARCH>]
+# ['aws-k8s-nvidia'] - [<FAMILY>-<FLAVOR>]
+# ['aws-k8s'.x86_64] - [<FAMILY>.<ARCH>]
+# ['aws-k8s'] - [<FAMILY>]
+# ['aws-nvidia'.x86_64] - [<PLATFORM>-<FLAVOR>.<ARCH>]
+# ['aws-nvidia'] - [<PLATFORM>-<FLAVOR>]
+# ['aws'.x86_64] - [<PLATFORM>.<ARCH>]
+# ['aws'] - [<PLATFORM>]
+# TestSys default values
+
+
+# Configuration for all AWS variants (platform level configuration).
+[aws]
+agent-role = "<AWS-TESTING-ROLE>"
+
+# Configuration for all nvidia AWS variants on x86_64 (platform-flavor level configuration)
+[aws-nvidia.x86_64]
+instance-type = "p3.2xlarge"
+
+# Configuration for all nvidia AWS variants on aarch64 (platform-flavor level configuration)
+[aws-nvidia.aarch64]
+instance-type = "g5g.2xlarge"
+
+# Configuration for all `aws-k8s` variants testing (family level configuration).
+[aws-k8s]
+# A single role can be assumed by agents to test all `aws-k8s` variants in a separate
+# testing account.
+agent-role = "arn:aws:iam::<ACCOUNT-ID>:role/<ROLE-NAME>"
+
+# The cluster name templating can be defined for all `aws-k8s` variants.
+# To test on ipv4 and ipv6 clusters, the following templates could be used.
+# Note: TestSys does not currently support creating ipv6 clusters, so the ipv6 cluster must already exist.
+cluster-names = ["{{arch}}-{{variant}}", "{{arch}}-{{variant}}-ipv6"]
+
+# A custom conformance registry may be needed for testing if image pull reliability is a concern.
+conformance-registry = "<ACCOUNT>.dkr.ecr.cn-north-1.amazonaws.com.cn"
```

Comment:
Does every key in these "configuration" sections correspond to a testsys command line argument? If so, could that be more explicitly described in the overview at the beginning?

Reply:
That is a rule with come cli's but that is not guaranteed here, for instance `testsys` can only use a single cluster, but multiple clusters can be described in `Test.toml`.